src/regex.nim

```diff
@@ -2100,6 +2101,51 @@ proc groupsCount*(m: RegexMatch): int =
   ##
   m.groups.len
 
+
+proc groupsNames*(m: RegexMatch): seq[string] =
+  ## return the names of capturing groups
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##     var m: RegexMatch
+  ##     doAssert "hello world".match("(?P<greet>hello) (?P<who>world)", m)
+  ##     doAssert m.groupsCount == 2
+
+
+  toSeq(m.namedGroups.keys)
+
+
+proc groupsByName*(m: RegexMatch, name: string, originalText: string): seq[string] =
+  ## return capturing groups by name
+  ##
+  ## .. code-block:: nim
+  ##   block:
+  ##     var m: RegexMatch
+  ##     doAssert "hello world".match("(?P<greet>hello) (?P<who>world)", m)
+  ##     doAssert m.groupByName("greet") == @["hello"]
+  ##     doAssert m.groupByName("who") == @["world"]
```

Comment:
Rename it to `group` and rename the `name` param to `groupName`. Description should say something like `Return all captures for a given capturing group`.